rest_framework_jwt/serializers.py

```diff
@@ -57,10 +57,25 @@ def validate(self, attrs):
 
                 payload = jwt_payload_handler(user)
 
-                return {
+                data = {
                     'token': jwt_encode_handler(payload),
```

Comment:
that's not correct.
jwt_encode_handler function should use JWT_SECRET defined on device if permanent token auth is used. Here it doesn't look like so because you create device (with wrong secret, btw) after this one.